### PR TITLE
Allow '/' to prefix container names to match Docker

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/16663

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
Podman now supports container names beginning with a '/' to match Docker behaviour.
```

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
I think there's more to unpack since Docker strips off the leading '/'.  '/foo' create the name "foo" and it will also be stripped of in regexes.  Yet `^/foo$` is not a valid regex to me.

**Comment by Luap99** [comment]
> I think there's more to unpack since Docker strips off the leading '/'. '/foo' create the name "foo" and it will also be stripped of in regexes. Yet `^/foo$` is not a valid regex to me.

docker always stores the name with a leading `/`, if you look at the API response they always include the slash. see the compat API for example https://github.com/containers/podman/blob/1b677c5202332ea7bb8753bd1fbe259c7323c991/pkg/api/handlers/compat/containers.go#L497
So if you apply a filter via docker api we must also add the slash to the name before running the filter regex.

File: cmd/podman/utils/utils.go [hunk 1]
```diff
@@ -139,3 +140,11 @@ func IsCheckpointImage(ctx context.Context, namesOrIDs []string) (bool, error) {
 	}
 	return true, nil
 }
+
+func RemoveSlash(input []string) []string {
+	var output []string
```
Comment (mheon) [inline cmd/podman/utils/utils.go:145]: Can initialize this with `make([]string, 0, len(input))` to ensure we only do one allocation

**Comment by rhatdan** [comment]
@vrothberg @Luap99 @mheon PTAL

**Comment by mheon** [comment]
Code LGTM

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Comment by rhatdan** [comment]
@vrothberg @giuseppe @flouthoc @ashley-cui PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

**Comment by vrothberg** [comment]
This is already in the v4.4 branch

**Pull request merged**